### DISCUSSION TITLE
Redirect

### DISCUSSION THREAD

**Discussion by Karlstens** [vuejs/vitepress#3340] [Q&A]
With this sample config.js, I'm needing the behaviour so that when the end user navigates to Shapes then clicks "Rectangle" in the sidebar, they're then taken to the Square page.

The problem I have is, I just can't have rectangle directly linking square because then sidebar lights up all the matching links that are active (both Rectangle and Square highlight). In my real world case, there's many (not just one) links in various sidebar locations, some are nested, and the sidebar essentially becomes confusing to the end user as all the text entries of the same link activate.

```js
import { defineConfig } from "vitepress";

export default defineConfig({
	title: "My VitePress Project",
	description: "Testing VitePress",
	
	themeConfig: {
		nav: [
			{ text: "Home", link: "/" },
			{ text: "Shapes", link: "/shapes" },
			{ text: "Colours", link: "/colours" },
			{ text: "Sizes", link: "/sizes" },
		],
		sidebar: {
			"/": [
				{
					text: "Shapes",
					collapsible: true,
					items: [
						{ text: "Overview", link: "/shapes" },
						{ text: "Circle", link: "/shapes/circle" },
						{ text: "Square", link: "/shapes/square" },
						// Need to redirect 'rectangle' to 'square', whilst URL reads "/shapes/rectangle" and sidebar has Rectangle highlighted. Then imagine, I have a dozen more similar entries, also all redirecting to square. 
						{ text: "Rectangle", link: "/shapes/rectangle" }, 
						{ text: "Triangle", link: "/shapes/triangle" },
					],
				},
				{
					text: "Colours",
					collapsible: true,
					items: [
						{ text: "Overview", link: "/colours" },
						{ text: "Red", link: "/colours/red" },
						{ text: "Green", link: "/colours/green" },
						{ text: "Blue", link: "/colours/blue" },
					],
				},
				{
					text: "Sizes",
					collapsible: true,
					items: [
						{ text: "Overview", link: "/sizes" },
						{ text: "Small", link: "/sizes/small" },
						{ text: "Medium", link: "/sizes/medium" },
						{ text: "Large", link: "/sizes/large" },
					],
				},
			],
		},
	},
});
```

I thought rewrite feature might be what I needed, but after testing, I don't think it is;

```js
...
rewrites: {
	"/shapes/rectangle": "/shapes/square"
},
...
```

With this, I was hoping the URL would perhaps continue to read /shapes/rectangle, whist serving /shapes/square content. 

Appreciate any help on this!

**Comment by Karlstens** [comment]
Through further reading, I've found an example under Dynamic Routes that looks to represent one method of achieving the navigational experience of what I'm after.

https://vitepress.dev/guide/routing#dynamic-routes

```js
//https://vitepress.dev/guide/routing#paths-loader-file

// docs\shapes\[square].md
// docs\shapes\[square].paths.js

//[square].paths.js
export default {
	paths() {
		return [
			{ params: { square: "square" } },
			{ params: { square: "large-square" } },
			{ params: { square: "rectangle" } }, 
		];
	},
};

//Builds the following;
//http://.../shapes/square.html
//http://.../shapes/large-square.html
//http://.../shapes/rectangle.html
```

For my example at least, I'm not sure if I'm comfortable with the idea of duplicating the same article again and again just to achieve a navigational experience - and how these duplicate articles will pan out in the bigger scheme of things. 

Keen to hear if other devs are achieving similar Sidebar Navigational experiences but by using different methods (without article duplication). 

PS: _I'm well aware that a square is a rectangle and that the reverse is not true_. I'll think of a better example for my next question. 😉

**Comment by HYP3R00T** [comment]
Why no one from the community has entertained this query? I am looking for a similar solution.
